### PR TITLE
Refactor base32 implementation

### PR DESCRIPTION
**Pull request by adolfogc**
## Description
- Improve code quality of `core/Base32.cpp`
- Remove deprecated `core/Optional.h`

## Motivation and context
Improve code-base.

## How has this been tested?
```bash
./tests/testbase32
./tests/testtotp
```

## Types of changes
- ✅ ~Bug~ fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by TheZ3ro** [comment]
Why using C uint instead of Qt quint?

**Comment by adolfogc** [comment]
@TheZ3ro Thanks for the review. You're right, it's better to keep using quint*N*. After reading closer http://en.cppreference.com/w/cpp/types/integer, I now see that uint*N*_t is not guaranteed to be present in all platforms, whereas Qts typedefs are. I'm reverting this change.

File: src/core/Base32.cpp [hunk 1]
```diff
@@ -195,14 +199,15 @@ QByteArray Base32::encode(const QByteArray& data)
             n = 20;
             break;
         default: // expand to 35 bits
-            Q_ASSERT(rBits == 32);
+            Q_ASSERT(32 == rBits);
```
Comment (weslly) [inline src/core/Base32.cpp]: Any particular reason for Yoda conditions? They look out of place since the rest of the project doesn't use it.

Comment (phoerious) [reply src/core/Base32.cpp]: I use them sometimes.

Comment (adolfogc) [reply src/core/Base32.cpp]: @weslly No other reason than personal preference. 

yoda-emoji Keep it in mind, I will.  My intention to differ from the project's codingstyle it is not. 😄

**Pull request merged**